google/resource_compute_instance.go

```diff
@@ -1506,6 +1530,12 @@ func expandBootDisk(d *schema.ResourceData, config *Config, zone *compute.Zone,
 		}
 	}
 
+	if v, ok := d.GetOk("boot_disk.0.kms_key_self_link"); ok {
```

Comment:
In this case because there's only ever one boot disk, we can add `ConflictsWith` in the schema (so you'd put `ConflictsWith: "[]string{"boot_disk.0.disk_encryption_key_raw"},` in the boot disk kms key section